packages/docgen/lib/get-jsdoc-from-token.js

```diff
@@ -25,10 +25,15 @@ module.exports = ( token ) => {
 			spacing: 'preserve',
 		} )[ 0 ];
 		if ( jsdoc ) {
+			let paramCount = 0;
 			jsdoc.tags = jsdoc.tags.map( ( tag ) => {
+				const isUnqualifiedParam =
+					tag.tag === 'param' && ! tag.name.includes( '.' );
+				const index = isUnqualifiedParam ? paramCount++ : paramCount;
```

Comment:
does TSdoc enforce that doc params have to be in the exact same order as the array params?

Reply:
I'm not sure if TSDoc enforces that but our automatic JSDoc generation that ESLint does will put it in the right order or complain if it's out of order.